src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -164,41 +170,44 @@ private void VerifyConnected()
 				// need to try IP Addresses one at a time: https://github.com/dotnet/corefx/issues/5829
 				foreach (var ipAddress in ipAddresses)
 				{
-					cancellationToken.ThrowIfCancellationRequested();
-
-					Socket socket = null;
+					var socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
 					try
 					{
-						socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
-						using (cancellationToken.Register(() => socket.Dispose()))
+						var connectTasks = new List<Task>
```

Comment:
I can't remember exactly - I'd have to go back and re-test.  Does an `await` command called on `socket.OpenAsync` without a cancellationToken block and not consider the handle on the CancellationToken?